contrib/ansible/create-cluster-playbook.yaml

```diff
@@ -82,25 +84,22 @@
       path: "{{ playbook_dir }}/../../certs/"
       state: directory
 
-  - name: generate cluster privatekey
-    openssl_privatekey:
-      path: "{{ cluster_privatekey_path }}"
-
-  - name: generate cluster CSR
-    openssl_csr:
-      path: "{{ cluster_csr_path }}"
-      privatekey_path: "{{ cluster_privatekey_path }}"
-      # TODO: these were previously junk (re-using our apiserver certs), but can we do something meaningful here?
-      common_name: "{{ cluster_name }}.example.com"
-      #subject_alt_name:
-        #- "{{ cluster_name }}"
-
-  - name: generate cluster cert
-    openssl_certificate:
-      path: "{{ cluster_cert_path }}"
-      privatekey_path: "{{ cluster_privatekey_path }}"
-      csr_path: "{{ cluster_csr_path }}"
-      provider: selfsigned
+  - include_role:
+      name: cluster_operator_certgen
+    vars:
+      cert_dir: "{{ cluster_cert_dir }}"
+      ca_name: "{{ cluster_ca_name }}"
+      cert_name: "{{ cluster_name }}"
+      # TODO: do something meaningful here?
+      cn: "{{ cluster_name }}"
+      alt_names: "{{ cluster_name }}.{{ cluster_namespace }}\",\"{{ cluster_name }}.{{ cluster_namespace }}.svc"
```

Comment:
How does this work? If I expand this, I would get a string like:
`cewong-cluster.myproject","cewong-cluster.myproject.svc`

Reply:
Pretty sure it's all just junk today, this is just modeled on how it was working before, and we were actually using the apiserver's cert in any case. Definitely needs to be looked at so added the TODO for both this and the CN.